cmd/context/selector.go

```diff
@@ -0,0 +1,402 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"text/template"
+
+	"github.com/chzyer/readline"
+	"github.com/juju/ansiterm"
+	"github.com/manifoldco/promptui"
+	"github.com/manifoldco/promptui/list"
+	"github.com/manifoldco/promptui/screenbuf"
+	"github.com/okteto/okteto/pkg/log"
+)
+
+const (
+	esc        = "\033["
+	hideCursor = esc + "?25l"
+	showCursor = esc + "?25h"
+	clearLine  = esc + "2K"
+)
+
+type OktetoSelector struct {
+	Label string
+	Items []SelectorItem
+	Size  int
+
+	Templates *promptui.SelectTemplates
+	Keys      *promptui.SelectKeys
+	list      *list.List
+
+	OktetoTemplates *OktetoTemplates
+}
+
+type OktetoTemplates struct {
+	FuncMap  template.FuncMap
+	label    *template.Template
+	active   *template.Template
+	inactive *template.Template
+	selected *template.Template
+	details  *template.Template
+	help     *template.Template
+}
+
+type SelectorItem struct {
+	Label  string
+	Enable bool
+}
+
+func getContextsSelection(ctxOptions *ContextOptions) []SelectorItem {
+	clusters := []SelectorItem{
+		{
+			Label:  "Okteto contexts",
+			Enable: false,
+		},
+		{
+			Label:  "Okteto Cloud",
+			Enable: true,
+		},
+		{
+			Label:  "Okteto Enterprise",
+			Enable: true,
+		},
+	}
+	if !ctxOptions.OnlyOkteto {
+		k8sClusters := getKubernetesContextList()
+		if len(k8sClusters) > 0 {
+			clusters = append(clusters, SelectorItem{
+				Label:  "Kubernetes contexts",
+				Enable: false,
+			})
+			for _, k8sCluster := range k8sClusters {
+				clusters = append(clusters, SelectorItem{
+					Label:  k8sCluster,
+					Enable: true,
+				})
+			}
+		}
+	}
+	return clusters
+}
+
+func AskForOptions(options []SelectorItem, label string) (string, error) {
+	prompt := OktetoSelector{
+		Label: label,
+		Items: options,
+		Size:  len(options),
+		Templates: &promptui.SelectTemplates{
+			Label:    "{{ .Label }}",
+			Selected: "{{if .Enable}} ✓  {{ .Label | oktetoblue }}{{else}}{{ .Label | oktetoblue}}{{end}}",
+			Active:   fmt.Sprintf("{{if .Enable}}%s {{ .Label | oktetoblue }}{{else}}{{ .Label | oktetoblue}}{{end}}", promptui.IconSelect),
+			Inactive: "{{if .Enable}}  {{ .Label | oktetoblue}}{{else}}{{ .Label | oktetoblue}}{{end}}",
+			FuncMap:  promptui.FuncMap,
+		},
+	}
+
+	prompt.Templates.FuncMap["oktetoblue"] = log.BlueString
+
+	optionSelected, err := prompt.Run()
+	if err != nil {
+		log.Infof("invalid init option: %s", err)
+		return "", fmt.Errorf("invalid option")
+	}
+
+	return optionSelected, nil
+}
+
+func (s OktetoSelector) Run() (string, error) {
```

Comment:
Did you test it on windows?

Reply:
I will try it and send some screenshots from different windows terminals

Reply:
From left to right (powershell, cmd, git cmd and windows terminal) :
 <img width="829" alt="Captura de pantalla 2021-10-01 175818" src="https://user-images.githubusercontent.com/25170843/135651453-3ba5b0a2-8e8c-43fc-aec1-7474bac211ba.png">